### PR TITLE
hugo-links-move-for-4.10-community

### PR DESCRIPTION
**Pull request by colin-hillis**
All remaining relative links preceded by ../ have been updated to use Hugo links for 4.10 Community:

Approximately 100 links or slightly more in total, across approx 80 files, with 18-20 broken links identified by the Hugo link checker.

### CODE REVIEW

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md [hunk 1]
```diff
@@ -374,7 +374,7 @@ To check if the CorDapp is signed, use the [JAR signing and verification tool](h
 jarsigner --verify path/to/cordapp.jar
 ```
 
-The Cordformation plugin can also sign CorDapp JARs when [deploying a set of nodes](generating-a-node.md).
+The Cordformation plugin can also sign CorDapp JARs when [deploying a set of nodes](./generating-a-node.md).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/cordapp-build-systems.md:377]: Should this be relref too?

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md [hunk 2]
```diff
@@ -35,9 +35,9 @@ Code samples guide you at every step.
 
 You will need to:
 
-* Know [what a CorDapp is](../../../../../../en/platform/corda/4.8/enterprise/cordapps/cordapp-overview.md).
-* Set up your [development environment](../../../../../../en/platform/corda/4.8/enterprise/cordapps/getting-set-up.md).
-* Run a [sample CorDapp](../../../../../../en/platform/corda/4.8/enterprise/cordapps/tutorial-cordapp.md) to see Corda in action (optional).
+* Know [what a CorDapp is]({{< relref "../../../../../../en/platform/corda/4.8/enterprise/cordapps/cordapp-overview.md" >}}).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/cordapp-build-systems.md:38]:
```suggestion
* Know [what a CorDapp is]({{< relref "../enterprise/cordapps/cordapp-overview.md" >}}).
```

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md [hunk 3]
```diff
@@ -35,9 +35,9 @@ Code samples guide you at every step.
...
+* Set up your [development environment]({{< relref "../../../../../../en/platform/corda/4.8/enterprise/cordapps/getting-set-up.md" >}}).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/cordapp-build-systems.md:39]:
```suggestion
* Set up your [development environment]({{< relref "../enterprise/cordapps/getting-set-up.md" >}}).
```

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md [hunk 4]
```diff
@@ -35,9 +35,9 @@ Code samples guide you at every step.
...
+* Run a [sample CorDapp]({{< relref "../../../../../../en/platform/corda/4.8/enterprise/cordapps/tutorial-cordapp.md" >}}) to see Corda in action (optional).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/cordapp-build-systems.md:40]:
```suggestion
* Run a [sample CorDapp]({{< relref "../enterprise/cordapps/tutorial-cordapp.md" >}}) to see Corda in action (optional).
```

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md [hunk 5]
```diff
@@ -248,7 +248,7 @@ JAR, and not different versions of the JAR created from identical sources.
 The filename of the JAR must include a unique identifier to deduplicate it from other releases of the same CorDapp.
 This is typically done by appending the version string to the CorDapp’s name. This unique identifier should not change
 once the JAR has been deployed on a node. If it does, make sure no one is relying on `FlowContext.appName` in their
-flows (see [Versioning](../../../../../../en/platform/corda/4.8/enterprise/cordapps/versioning.md)).
+flows (see [Versioning]({{< relref "../../../../../../en/platform/corda/4.8/enterprise/cordapps/versioning.md" >}})).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/cordapp-build-systems.md:251]:
```suggestion
flows (see [Versioning]({{< relref "../enterprise/cordapps/versioning.md" >}})).
```

File: content/en/platform/corda/4.10/community/getting-set-up.md [hunk 6]
```diff
@@ -24,7 +24,7 @@ There are four pieces of required software for CorDapp development:
 * Git
 * Gradle
 
-{{< note >}} Visit the [platform support matrix](../../../../../en/platform/corda/4.8/enterprise/platform-support-matrix.md) to check the latest third-party application versions supported by Corda. {{< /note >}}
+{{< note >}} Visit the [platform support matrix]({{< relref "../../../../../en/platform/corda/4.8/enterprise/platform-support-matrix.md" >}}) to check the latest third-party application versions supported by Corda. {{< /note >}}
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/getting-set-up.md:27]:
```suggestion
{{< note >}} Visit the [platform support matrix]({{< relref "release-platform-support-matrix.md" >}}) to check the latest third-party application versions supported by Corda. {{< /note >}}
```

File: content/en/platform/corda/4.10/community/secure-coding-guidelines.md [hunk 7]
```diff
@@ -62,5 +62,5 @@ Make sure your contracts are secure. Check that:
 
 ## Related Content
 Learn more about:
-* [Writing flows](../../../../../en/platform/corda/4.8/enterprise/flow-state-machines.md)
+* [Writing flows]({{< relref "../../../../../en/platform/corda/4.8/enterprise/flow-state-machines.md" >}})
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/secure-coding-guidelines.md:65]:
```suggestion
* [Writing flows]({{< relref "../enterprise/flow-state-machines.md" >}})
```

File: content/en/platform/corda/4.10/community/tutorial-cordapp.md [hunk 8]
```diff
@@ -271,7 +271,7 @@ You can see the newly-created IOU by running `run vaultQuery contractStateType:
 
 ### Via the h2 web console
 
-You can connect directly to your node’s database to see its stored states, transactions, and attachments. Follow the instructions in [Node database](../../4.8/enterprise/node/operating/node-database.md).
+You can connect directly to your node’s database to see its stored states, transactions, and attachments. Follow the instructions in [Node database]({{< relref "../../4.8/enterprise/node/operating/node-database.md" >}}).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/tutorial-cordapp.md:274]:
```suggestion
You can connect directly to your node’s database to see its stored states, transactions, and attachments. Follow the instructions in [Node database]({{< relref "../enterprise/node/operating/node-database.md" >}}).
```

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

**Review by yarivus — CHANGES_REQUESTED** [review]
@colin-hillis 
I see several inconsistencies: 

1. You use relref but don't remove the .md (as observed in other PRs)
2. Some of the links next to the changes were not changed using relref (@paulmoloneyr3 highlighted some in his comments)
3. Following another PR review earlier - we reference enterprise links while in the community pages (?!)

File: content/en/platform/corda/4.10/community/cordapp-build-systems.md (hunk 1 above)
Comment (colin-hillis) [reply content/en/platform/corda/4.10/community/cordapp-build-systems.md:377]: Good spot. This was a manual edit.

**Comment by colin-hillis** [comment]
> @colin-hillis I see several inconsistencies:
> 
> 1. You use relref but don't remove the .md (as observed in other PRs)
> 2. Some of the links next to the changes were not changed using relref (@paulmoloneyr3 highlighted some in his comments)
> 3. Following another PR review earlier - we reference enterprise links while in the community pages (?!)

1. So the ones missing .md in the very first merge for one single page....that was my mistake to not include .md. It actually makes no difference, but it is more a consistency thing.
2. Paul's changes to the overuse of ../../ are not in the scope of this. This is to convert the link types, but the use of far too many ../s has already been highlighted as a follow up project in the Confluence page for this work. So Paul's changes (although I have accepted them), are not in scope.
3. This was the focus of a previous effort by Paul, but lots of links were missed because we didn't have a reliable link checker. I identified 16-18 earlier. So this roll out is helping the problem.
4. Additionally the scope of this above commits was for all links starting ../ A further commit will follow to deal with links starting ./. In fact there are several other follow ups to get everything right (all of which are on the confluence page), but this is 100% iterative. So just ../ links first.

It is far too risky to try to do it all at once.

**Comment by colin-hillis** [comment]
For clarity on scope, see: https://r3-cev.atlassian.net/wiki/spaces/TW/pages/4390879344/Hugo+Links#Steps-for-Changes

**Comment by yarivus** [comment]
> > @colin-hillis I see several inconsistencies:
> > 
> > 1. You use relref but don't remove the .md (as observed in other PRs)
> > 2. Some of the links next to the changes were not changed using relref (@paulmoloneyr3 highlighted some in his comments)
> > 3. Following another PR review earlier - we reference enterprise links while in the community pages (?!)
> 
> 1. So the ones missing .md in the very first merge for one single page....that was my mistake to not include .md. It actually makes no difference, but it is more a consistency thing.
> 2. Paul's changes to the overuse of ../../ are not in the scope of this. This is to convert the link types, but the use of far too many ../s has already been highlighted as a follow up project in the Confluence page for this work. So Paul's changes (although I have accepted them), are not in scope.
> 3. This was the focus of a previous effort by Paul, but lots of links were missed because we didn't have a reliable link checker. I identified 16-18 earlier. So this roll out is helping the problem.
> 4. Additionally the scope of the above commits was for all links starting ../ A further commit will follow to deal with links starting ./. In fact there are several other follow ups to get everything right (all of which are on the confluence page), but this is 100% iterative. I initially wanted to just do ../, but I will do ./ now too. The other 6 or 7 links types and edge cases will have to wait though for the next effort.
> 
> It is far too risky to try to do them all at once.

@colin-hillis there are still 'community' pages referencing 'enterprise' links.
I haven't understood from your reply whether that is intended, and if not what's the plan of fixing this inconsistency ?
Is the plan to do it in a following PR ?

**Comment by colin-hillis** [comment]
Hi Yariv,

Whilst some were identified and fixed because of they were just 404s, if some of Paul's links between Community and Enterprise are actually correct links to begin with, this effort of mine is not going to spot them. That is out of scope.

The purpose of this process is to migrate to the recommended Hugo links and some existing broken links will naturally be fixed along the way. But the correction of already working links is not going to be solved by this. However, I have another idea for Paul to try, which will find all instances of the mentioned cross-links and allow them to be fixed.  I can talk to @paulmoloneyr3 about this separately.

**Comment by yarivus** [comment]
> Hi Yariv,
> 
> Whilst some were identified and fixed because of they were just 404s, if some of Paul's links between Community and Enterprise are actually correct links to begin with, this effort of mine is not going to spot them. That is out of scope.
> 
> The purpose of this process is to migrate to the recommended Hugo links and some existing broken links will naturally be fixed along the way. But the correction of already working links is not going to be solved by this. However, I have another idea for Paul to try, which will find all instances of the mentioned cross-links and allow them to be fixed. I can talk to @paulmoloneyr3 about this separately.

OK - I understand that the community pages referencing enterprise is out of scope here for this PR.
@paulmoloneyr3 - please create a separate PR for that problem.

**Review by yarivus — APPROVED** [review]
LGTM

**Comment by colin-hillis** [comment]
@paulmoloneyr3 once I am back, let's chat about this before going ahead please. It has been like this for weeks, so another day will be ok. Cheers Paul.

**Comment by paulmoloneyr3** [comment]
> @paulmoloneyr3 once I am back, let's chat about this before going ahead please. It has been like this for weeks, so another day will be ok. Cheers Paul.

Talk to you then!

**Review by paulmoloneyr3 — APPROVED** [review]

**Pull request merged**